overlord/snapstate/snapmgr.go

```diff
@@ -209,6 +209,22 @@ func (snapst *SnapState) Block() []snap.Revision {
 
 var ErrNoCurrent = errors.New("snap has no current revision")
 
+// Retrieval functions
+var readInfo = readInfoImpl
+
+func readInfoImpl(name string, si *snap.SideInfo) (*snap.Info, error) {
+	info, err := snap.ReadInfo(name, si)
+	if _, ok := err.(*snap.NotFoundError); ok {
+		reason := fmt.Sprintf("cannot read snap info: %s", err)
```

Comment:
s/ info// - "info" is an implementation detail here, for users we couldn't read the actual snap.